draft-spencer-haapi/draft-spencer-haapi-protocol.md

```diff
@@ -0,0 +1,308 @@
+
+For these reasons and others, an alternative protocol is needed that can address these challenges. Such a protocol is described below.
+
+### 1.1. Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14 ([RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119)) ([RFC 8174](https://datatracker.ietf.org/doc/html/rfc8174)) when, and only when, they appear in all capitals, as shown here.
+
+This specification uses the terms "access token", "refresh token", "authorization server", "resource server", "authorization endpoint", "authorization request", "authorization response", "token endpoint", "grant type", "access token request", "access token response", and "client" defined by The OAuth 2.0 Authorization Framework ([RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749)).
+
+Additionally, this specification defines and uses the following terms:
+
+1. HAAPI    - Hypermedia Authentication API
+2. CAT      - Client Attestation Token
+3. SPA      - Single Page Application
+
+## 2. Protocol Overview
+
+An authorization server (in the OAuth and OpenID Connect sense of the term) supporting this specification provides an REST API that authenticates users operating attested client applications. This interaction is shown in figure 1:
+
+```text
++--------------------+                                   Token   +--------------------+
+|                    |              CAT                  endpoint|                    |
+|                    +--(A)---------------------------->o--------+                    |
+|                    |                                           |                    |
+|                    |<-(B)--------------------------------------+                    |
+|                    |  API AT                                   |                    |
+|                    |                              Authorization|                    |
+|                    |   API AT & PoP                    endpoint|                    |
+|                    +--(C)---------------------------->o--------+                    |
+|                    |                                           |                    |
+|                    |                             Authentication|                    |
+|       OAuth        |   API AT & PoP              endpoint      |    Authorization   |
+|      Client        +--(D)---------------------------->o--------+       Server       |
```

Comment:
Should the use of the authentication endpoint, and later the LT part of the specification/example? It can also be an implementation detail of Curity?

Reply:
I was on the fence on this part. Will remove.